test/test_store/test_store_auditable.py

```diff
@@ -1,386 +1,472 @@
-            [
-                (EX.s1, EX.p1, EX.o1),
-                (EX.s0, EX.p0, EX.o0bis),
-            ],
-        )
-
-    def test_rollback_commit(self):
-        self.t1.rollback()
-        self.t2.commit()
-        self.assert_graph_equal(
-            self.g,
-            [
-                (EX.s0, EX.p0, EX.o0),
-                (EX.s2, EX.p2, EX.o2),
-            ],
-        )
-
-    def test_rollback_rollback(self):
-        self.t1.rollback()
-        self.t2.rollback()
-        self.assert_graph_equal(
-            self.g,
-            [
-                (EX.s0, EX.p0, EX.o0),
-                (EX.s0, EX.p0, EX.o0bis),
-            ],
-        )
-
-
-class TestAuditableStoreEmbeded(BaseTestAuditableStore):
-    def setUp(self):
-        self.g = Graph()
-        self.g.add((EX.s0, EX.p0, EX.o0))
-        self.g.add((EX.s0, EX.p0, EX.o0bis))
-
-        self.t1 = Graph(AuditableStore(self.g.store), self.g.identifier)
-        self.t1.add((EX.s1, EX.p1, EX.o1))
-        self.t1.remove((EX.s0, EX.p0, EX.o0bis))
-
-        self.t2 = Graph(AuditableStore(self.t1.store), self.t1.identifier)
-        self.t2.add((EX.s2, EX.p2, EX.o2))
-        self.t2.remove((EX.s1, EX.p1, EX.o1))
-
-    def test_commit_commit(self):
-        self.assert_graph_equal(
-            self.t2,
-            [
-                (EX.s0, EX.p0, EX.o0),
-                (EX.s2, EX.p2, EX.o2),
-            ],
-        )
-        self.t2.commit()
-        self.assert_graph_equal(
-            self.t1,
-            [
-                (EX.s0, EX.p0, EX.o0),
-                (EX.s2, EX.p2, EX.o2),
-            ],
-        )
-        self.t1.commit()
-        self.assert_graph_equal(
-            self.g,
-            [
-                (EX.s0, EX.p0, EX.o0),
-                (EX.s2, EX.p2, EX.o2),
-            ],
-        )
-
-    def test_commit_rollback(self):
-        self.t2.commit()
-        self.t1.rollback()
-        self.assert_graph_equal(
-            self.g,
-            [
-                (EX.s0, EX.p0, EX.o0),
-                (EX.s0, EX.p0, EX.o0bis),
-            ],
-        )
-
-    def test_rollback_commit(self):
-        self.t2.rollback()
-        self.assert_graph_equal(
-            self.t1,
-            [
-                (EX.s0, EX.p0, EX.o0),
-                (EX.s1, EX.p1, EX.o1),
-            ],
-        )
-        self.t1.commit()
-        self.assert_graph_equal(
-            self.g,
-            [
-                (EX.s0, EX.p0, EX.o0),
-                (EX.s1, EX.p1, EX.o1),
-            ],
-        )
-
-    def test_rollback_rollback(self):
-        self.t2.rollback()
-        self.t1.rollback()
-        self.assert_graph_equal(
-            self.g,
-            [
-                (EX.s0, EX.p0, EX.o0),
-                (EX.s0, EX.p0, EX.o0bis),
-            ],
-        )
+            graph = Graph(store=s.name)
+            pluginstores.append(s.name)
+        except ImportError:
+            pass
+    return pluginstores
```

Comment:
I'm not entirely sure why this is here, the end result is this:


```
$ .venv/bin/python3 -m pytest test/test_store/test_store_auditable.py  --log-level DEBUG -rA --tb=native -v
============================================================================ test session starts ============================================================================
platform linux -- Python 3.9.12, pytest-7.1.1, pluggy-1.0.0 -- /home/iwana/sw/d/github.com/iafork/rdflib.cleanish/.venv/bin/python3
cachedir: .pytest_cache
rootdir: /home/iwana/sw/d/github.com/iafork/rdflib.cleanish, configfile: tox.ini
plugins: subtests-0.7.0, cov-3.0.0
collected 24 items                                                                                                                                                          

test/test_store/test_store_auditable.py::test_add_commit[BerkeleyDB] PASSED                                                                                           [  4%]
test/test_store/test_store_auditable.py::test_remove_commit[BerkeleyDB] PASSED                                                                                        [  8%]
test/test_store/test_store_auditable.py::test_multiple_remove_commit[BerkeleyDB] PASSED                                                                               [ 12%]
test/test_store/test_store_auditable.py::test_noop_add_commit[BerkeleyDB] PASSED                                                                                      [ 16%]
test/test_store/test_store_auditable.py::test_noop_remove_commit[BerkeleyDB] PASSED                                                                                   [ 20%]
test/test_store/test_store_auditable.py::test_add_remove_commit[BerkeleyDB] PASSED                                                                                    [ 25%]
test/test_store/test_store_auditable.py::test_remove_add_commit[BerkeleyDB] PASSED                                                                                    [ 29%]
test/test_store/test_store_auditable.py::test_add_rollback[BerkeleyDB] PASSED                                                                                         [ 33%]
test/test_store/test_store_auditable.py::test_remove_rollback[BerkeleyDB] PASSED                                                                                      [ 37%]
test/test_store/test_store_auditable.py::test_multiple_remove_rollback[BerkeleyDB] PASSED                                                                             [ 41%]
test/test_store/test_store_auditable.py::test_noop_add_rollback[BerkeleyDB] PASSED                                                                                    [ 45%]
test/test_store/test_store_auditable.py::test_noop_remove_rollback[BerkeleyDB] PASSED                                                                                 [ 50%]
test/test_store/test_store_auditable.py::test_add_remove_rollback[BerkeleyDB] PASSED                                                                                  [ 54%]
test/test_store/test_store_auditable.py::test_remove_add_rollback[BerkeleyDB] PASSED                                                                                  [ 58%]
test/test_store/test_store_auditable.py::test_add_commit_empty[BerkeleyDB] PASSED                                                                                     [ 62%]
test/test_store/test_store_auditable.py::test_add_rollback_empty[BerkeleyDB] PASSED                                                                                   [ 66%]
test/test_store/test_store_auditable.py::test_commit_commit PASSED                                                                                                    [ 70%]
test/test_store/test_store_auditable.py::test_commit_rollback PASSED                                                                                                  [ 75%]
test/test_store/test_store_auditable.py::test_rollback_commit PASSED                                                                                                  [ 79%]
test/test_store/test_store_auditable.py::test_rollback_rollback PASSED                                                                                                [ 83%]
test/test_store/test_store_auditable.py::test_commit_commit_embedded PASSED                                                                                           [ 87%]
test/test_store/test_store_auditable.py::test_commit_rollback_embedded PASSED                                                                                         [ 91%]
test/test_store/test_store_auditable.py::test_rollback_commit_embedded PASSED                                                                                         [ 95%]
test/test_store/test_store_auditable.py::test_rollback_rollback_embedded PASSED                                                                                       [100%]
```

On master branch these tests are not using BerkeleyDB, so I think something is different here. Ideally this should be limited to testing only auditable and not other stores.

Reply:
> Ideally this should be limited to testing only auditable and not other stores.

Well, okay. I won't push it if you think that's adequate. I'll trim down the test from the last commit so that it just runs in memory and I'll try to find another route for supporting store tests iterated over plugin stores.

Reply:
The way it is now looks good/